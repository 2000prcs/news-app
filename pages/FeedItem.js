import React from 'react';
import { object } from 'prop-types';
import {
  ListGroupItem,
  Media,
  Badge,
} from 'reactstrap';

const FeedItem = ({ news }) => {
  const imgStyle = {
    height: '160px',
    width: '160px',
    margin: '10px',
  };

  const textStyle = {
    fontStyle: 'italic',
  };

  return (
    // Render image from the news image_url (use default image for fallback if error occurs)
    <ListGroupItem>
      <Media>
        <Media left href={news.url} target="_blank">        
          <Media style={imgStyle} object src={news.image_url ? news.image_url : "../static/image.jpg"} onError={(e) => {e.target.src="../static/image.jpg"}} alt="news-image" /> 
        </Media>
        <Media body>
          <Media heading>
          <a href={news.url} target="_blank">{news.title}</a>
        </Media>
          {news.description}
          <br />
          <br />
          <span style={textStyle} >Source: <a href={news.site.url} target="_blank">{news.site.name}</a></span>
          <br />
          <span style={textStyle}>Published @ {news.published_at.substr(0, 10)}</span>
        </Media>
          <Badge color="secondary">News Score: {Math.ceil(news.score)}</Badge>
      </Media>
    </ListGroupItem>
  );
};

FeedItem.propTypes = {
  news: object.isRequired,
};

export default FeedItem;
