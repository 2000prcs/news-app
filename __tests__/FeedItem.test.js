import React from 'react';
import FeedItem from '../pages/FeedItem';
import { data } from '../static/data';

describe('FeedItem component', () => {
  test('Should render FeedItem', () => {
    const FeedItemWrapper = shallow(<FeedItem news={data[0]} />);
    expect(FeedItemWrapper).toMatchSnapshot();
  });
});
