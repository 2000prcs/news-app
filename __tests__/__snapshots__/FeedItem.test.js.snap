// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedItem component Should render FeedItem 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FeedItem
    news={
      Object {
        "authors": null,
        "created": "2018-06-21T18:29:50.104Z",
        "description": "Facebook, Amazon, Apple, Netflix and Google (FAANG) have been the motor of the S&P 500",
        "id": "NLEJ7gxOak",
        "image_url": "https://www.economist.com/assets/the-economist-logo.png",
        "language": "en",
        "mobile_url": null,
        "published_at": "2018-06-23T00:00:00.000Z",
        "read_time": 180,
        "score": 84.80139,
        "site": Object {
          "hostname": "www.economist.com",
          "id": "z28aeDEnRm",
          "language": "en",
          "mobile_url": null,
          "name": "The Economist",
          "slug": null,
          "url": "http://www.economist.com",
        },
        "slug": "most-stockmarket-returns-come-from-a-tiny-fraction-of-shares-4212887524",
        "title": "Most stockmarket returns come from a tiny fraction of shares",
        "url": "https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares",
        "word_count": 814,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Media>
        <Media
          href="https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares"
          left={true}
          target="_blank"
        >
          <Media
            alt="news-image"
            object={true}
            onError={[Function]}
            src="https://www.economist.com/assets/the-economist-logo.png"
            style={
              Object {
                "height": "160px",
                "margin": "10px",
                "width": "160px",
              }
            }
          />
        </Media>
        <Media
          body={true}
        >
          <Media
            heading={true}
          >
            <a
              href="https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares"
              target="_blank"
            >
              Most stockmarket returns come from a tiny fraction of shares
            </a>
          </Media>
          Facebook, Amazon, Apple, Netflix and Google (FAANG) have been the motor of the S&P 500
          <br />
          <br />
          <span
            style={
              Object {
                "fontStyle": "italic",
              }
            }
          >
            Source: 
            <a
              href="http://www.economist.com"
              target="_blank"
            >
              The Economist
            </a>
          </span>
          <br />
          <span
            style={
              Object {
                "fontStyle": "italic",
              }
            }
          >
            Published @ 
            2018-06-23
          </span>
        </Media>
        <Badge
          color="secondary"
          pill={false}
          tag="span"
        >
          News Score: 
          85
        </Badge>
      </Media>,
      "tag": "li",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Media
            href="https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares"
            left={true}
            target="_blank"
          >
            <Media
              alt="news-image"
              object={true}
              onError={[Function]}
              src="https://www.economist.com/assets/the-economist-logo.png"
              style={
                Object {
                  "height": "160px",
                  "margin": "10px",
                  "width": "160px",
                }
              }
            />
          </Media>,
          <Media
            body={true}
          >
            <Media
              heading={true}
            >
              <a
                href="https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares"
                target="_blank"
              >
                Most stockmarket returns come from a tiny fraction of shares
              </a>
            </Media>
            Facebook, Amazon, Apple, Netflix and Google (FAANG) have been the motor of the S&P 500
            <br />
            <br />
            <span
              style={
                Object {
                  "fontStyle": "italic",
                }
              }
            >
              Source: 
              <a
                href="http://www.economist.com"
                target="_blank"
              >
                The Economist
              </a>
            </span>
            <br />
            <span
              style={
                Object {
                  "fontStyle": "italic",
                }
              }
            >
              Published @ 
              2018-06-23
            </span>
          </Media>,
          <Badge
            color="secondary"
            pill={false}
            tag="span"
          >
            News Score: 
            85
          </Badge>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Media
              alt="news-image"
              object={true}
              onError={[Function]}
              src="https://www.economist.com/assets/the-economist-logo.png"
              style={
                Object {
                  "height": "160px",
                  "margin": "10px",
                  "width": "160px",
                }
              }
            />,
            "href": "https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares",
            "left": true,
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": "news-image",
              "object": true,
              "onError": [Function],
              "src": "https://www.economist.com/assets/the-economist-logo.png",
              "style": Object {
                "height": "160px",
                "margin": "10px",
                "width": "160px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "body": true,
            "children": Array [
              <Media
                heading={true}
              >
                <a
                  href="https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares"
                  target="_blank"
                >
                  Most stockmarket returns come from a tiny fraction of shares
                </a>
              </Media>,
              "Facebook, Amazon, Apple, Netflix and Google (FAANG) have been the motor of the S&P 500",
              <br />,
              <br />,
              <span
                style={
                  Object {
                    "fontStyle": "italic",
                  }
                }
              >
                Source: 
                <a
                  href="http://www.economist.com"
                  target="_blank"
                >
                  The Economist
                </a>
              </span>,
              <br />,
              <span
                style={
                  Object {
                    "fontStyle": "italic",
                  }
                }
              >
                Published @ 
                2018-06-23
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <a
                  href="https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares"
                  target="_blank"
                >
                  Most stockmarket returns come from a tiny fraction of shares
                </a>,
                "heading": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Most stockmarket returns come from a tiny fraction of shares",
                  "href": "https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "Most stockmarket returns come from a tiny fraction of shares",
                "type": "a",
              },
              "type": [Function],
            },
            "Facebook, Amazon, Apple, Netflix and Google (FAANG) have been the motor of the S&P 500",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Source: ",
                  <a
                    href="http://www.economist.com"
                    target="_blank"
                  >
                    The Economist
                  </a>,
                ],
                "style": Object {
                  "fontStyle": "italic",
                },
              },
              "ref": null,
              "rendered": Array [
                "Source: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "The Economist",
                    "href": "http://www.economist.com",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "The Economist",
                  "type": "a",
                },
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Published @ ",
                  "2018-06-23",
                ],
                "style": Object {
                  "fontStyle": "italic",
                },
              },
              "ref": null,
              "rendered": Array [
                "Published @ ",
                "2018-06-23",
              ],
              "type": "span",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "News Score: ",
              85,
            ],
            "color": "secondary",
            "pill": false,
            "tag": "span",
          },
          "ref": null,
          "rendered": Array [
            "News Score: ",
            85,
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Media>
          <Media
            href="https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares"
            left={true}
            target="_blank"
          >
            <Media
              alt="news-image"
              object={true}
              onError={[Function]}
              src="https://www.economist.com/assets/the-economist-logo.png"
              style={
                Object {
                  "height": "160px",
                  "margin": "10px",
                  "width": "160px",
                }
              }
            />
          </Media>
          <Media
            body={true}
          >
            <Media
              heading={true}
            >
              <a
                href="https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares"
                target="_blank"
              >
                Most stockmarket returns come from a tiny fraction of shares
              </a>
            </Media>
            Facebook, Amazon, Apple, Netflix and Google (FAANG) have been the motor of the S&P 500
            <br />
            <br />
            <span
              style={
                Object {
                  "fontStyle": "italic",
                }
              }
            >
              Source: 
              <a
                href="http://www.economist.com"
                target="_blank"
              >
                The Economist
              </a>
            </span>
            <br />
            <span
              style={
                Object {
                  "fontStyle": "italic",
                }
              }
            >
              Published @ 
              2018-06-23
            </span>
          </Media>
          <Badge
            color="secondary"
            pill={false}
            tag="span"
          >
            News Score: 
            85
          </Badge>
        </Media>,
        "tag": "li",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Media
              href="https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares"
              left={true}
              target="_blank"
            >
              <Media
                alt="news-image"
                object={true}
                onError={[Function]}
                src="https://www.economist.com/assets/the-economist-logo.png"
                style={
                  Object {
                    "height": "160px",
                    "margin": "10px",
                    "width": "160px",
                  }
                }
              />
            </Media>,
            <Media
              body={true}
            >
              <Media
                heading={true}
              >
                <a
                  href="https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares"
                  target="_blank"
                >
                  Most stockmarket returns come from a tiny fraction of shares
                </a>
              </Media>
              Facebook, Amazon, Apple, Netflix and Google (FAANG) have been the motor of the S&P 500
              <br />
              <br />
              <span
                style={
                  Object {
                    "fontStyle": "italic",
                  }
                }
              >
                Source: 
                <a
                  href="http://www.economist.com"
                  target="_blank"
                >
                  The Economist
                </a>
              </span>
              <br />
              <span
                style={
                  Object {
                    "fontStyle": "italic",
                  }
                }
              >
                Published @ 
                2018-06-23
              </span>
            </Media>,
            <Badge
              color="secondary"
              pill={false}
              tag="span"
            >
              News Score: 
              85
            </Badge>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Media
                alt="news-image"
                object={true}
                onError={[Function]}
                src="https://www.economist.com/assets/the-economist-logo.png"
                style={
                  Object {
                    "height": "160px",
                    "margin": "10px",
                    "width": "160px",
                  }
                }
              />,
              "href": "https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares",
              "left": true,
              "target": "_blank",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "alt": "news-image",
                "object": true,
                "onError": [Function],
                "src": "https://www.economist.com/assets/the-economist-logo.png",
                "style": Object {
                  "height": "160px",
                  "margin": "10px",
                  "width": "160px",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "body": true,
              "children": Array [
                <Media
                  heading={true}
                >
                  <a
                    href="https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares"
                    target="_blank"
                  >
                    Most stockmarket returns come from a tiny fraction of shares
                  </a>
                </Media>,
                "Facebook, Amazon, Apple, Netflix and Google (FAANG) have been the motor of the S&P 500",
                <br />,
                <br />,
                <span
                  style={
                    Object {
                      "fontStyle": "italic",
                    }
                  }
                >
                  Source: 
                  <a
                    href="http://www.economist.com"
                    target="_blank"
                  >
                    The Economist
                  </a>
                </span>,
                <br />,
                <span
                  style={
                    Object {
                      "fontStyle": "italic",
                    }
                  }
                >
                  Published @ 
                  2018-06-23
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <a
                    href="https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares"
                    target="_blank"
                  >
                    Most stockmarket returns come from a tiny fraction of shares
                  </a>,
                  "heading": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Most stockmarket returns come from a tiny fraction of shares",
                    "href": "https://www.economist.com/finance-and-economics/2018/06/23/most-stockmarket-returns-come-from-a-tiny-fraction-of-shares",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "Most stockmarket returns come from a tiny fraction of shares",
                  "type": "a",
                },
                "type": [Function],
              },
              "Facebook, Amazon, Apple, Netflix and Google (FAANG) have been the motor of the S&P 500",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Source: ",
                    <a
                      href="http://www.economist.com"
                      target="_blank"
                    >
                      The Economist
                    </a>,
                  ],
                  "style": Object {
                    "fontStyle": "italic",
                  },
                },
                "ref": null,
                "rendered": Array [
                  "Source: ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "The Economist",
                      "href": "http://www.economist.com",
                      "target": "_blank",
                    },
                    "ref": null,
                    "rendered": "The Economist",
                    "type": "a",
                  },
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Published @ ",
                    "2018-06-23",
                  ],
                  "style": Object {
                    "fontStyle": "italic",
                  },
                },
                "ref": null,
                "rendered": Array [
                  "Published @ ",
                  "2018-06-23",
                ],
                "type": "span",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                "News Score: ",
                85,
              ],
              "color": "secondary",
              "pill": false,
              "tag": "span",
            },
            "ref": null,
            "rendered": Array [
              "News Score: ",
              85,
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
